ord_schema/updates.py

```diff
@@ -146,6 +157,38 @@ def update_reaction(reaction):
         event = reaction.provenance.record_modified.add()
         event.time.value = datetime.datetime.utcnow().ctime()
         event.details = 'Automatic updates from the submission pipeline.'
+    return id_substitutions
+
+
+def update_dataset(dataset):
+    """Updates a Dataset message.
+
+    Current updates:
+      * All reaction-level updates in update_reaction.
+      * reaction_id cross-references between Reactions in the dataset.
+
+    Args:
+        dataset: dataset_pb2.Dataset message.
+    """
+    # Reaction-level updates
+    id_substitutions = {}
+    for reaction in dataset.reactions:
+        id_substitutions.update(update_reaction(reaction))
+    # Dataset-level updates of cross-references
+    # Note: if the Dataset has been validated, then we know that all
+    # cross-referened reaction_ids correspond to placeholder reaction_ids of
+    # another Reaction in the Dataset. If not validated, this update might
+    # raise a KeyError exception.
```

Comment:
Please add a `Raises` section to the docstring; http://google.github.io/styleguide/pyguide.html#doc-function-raises